From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: amaanq <git@amaanq.com>
Date: Mon, 10 Feb 2026 00:00:00 +0000
Subject: [PATCH 158/162] disable HIBERNATION by default

Disable suspend-to-disk by default. Hibernation writes a full memory
image to disk and has a complex resume parser, both of which expand
the attack surface. Servers don't need it, and laptop users who want
it can re-enable CONFIG_HIBERNATION=y.

Signed-off-by: amaanq <git@amaanq.com>
---
 kernel/power/Kconfig | 1 +
 1 file changed, 1 insertion(+)

diff --git a/kernel/power/Kconfig b/kernel/power/Kconfig
index 05337f437cca..499478f86291 100644
--- a/kernel/power/Kconfig
+++ b/kernel/power/Kconfig
@@ -38,6 +38,7 @@
 config HIBERNATION
 	bool "Hibernation (aka 'suspend to disk')"
 	depends on SWAP && ARCH_HIBERNATION_POSSIBLE
+	default n
 	select HIBERNATE_CALLBACKS
 	select CRC32
 	select CRYPTO
-- 
2.48.1

